fl4health/losses/deep_mmd_loss.py

```diff
@@ -0,0 +1,224 @@
+from typing import Optional, Tuple
+
+import numpy as np
+import torch
+
+
+class ModelLatentF(torch.nn.Module):
+    """Latent space for both domains."""
+
+    def __init__(self, x_in: int, H: int, x_out: int):
+        """Init latent features."""
+        super(ModelLatentF, self).__init__()
+        self.restored = False
+        self.latent = torch.nn.Sequential(
+            torch.nn.Linear(x_in, H, bias=True),
+            torch.nn.Softplus(),
+            torch.nn.Linear(H, H, bias=True),
+            torch.nn.Softplus(),
+            torch.nn.Linear(H, H, bias=True),
+            torch.nn.Softplus(),
+            torch.nn.Linear(H, x_out, bias=True),
+        )
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        """Forward the LeNet."""
+        fealant = self.latent(input)
+        return fealant
+
+
+class DeepMmdLoss(torch.nn.Module):
+    def __init__(
+        self,
+        device: torch.device,
+        input_size: int,
+        lr: float = 0.001,
+        hidden_size: int = 10,
+        output_size: int = 50,
+        layer_name: Optional[str] = None,
+        training: bool = True,
+    ) -> None:
+        super().__init__()
+        self.device = device
+        self.lr = lr
+        self.layer_name = layer_name
```

Comment:
Any utility to the `layer_name` here? It seems like it's not used, but I may be missing something.